torchhd/datasets/abalone.py

```diff
@@ -0,0 +1,165 @@
+import os
+import os.path
+from typing import Callable, Optional, Tuple, List
+import torch
+from torch.utils import data
+import pandas as pd
+import tarfile
+import numpy as np
+
+from .utils import download_file
+
+
+class Abalone(data.Dataset):
+    """`Abalone <https://archive.ics.uci.edu/ml/datasets/abalone>`_ dataset.
+
+    Args:
+        root (string): Root directory containing the files of the dataset.
+        train (bool, optional): If True, returns training (sub)set stored in ``train.data`` as further determined by fold_id and fold_train variables.
+            Otherwise tries to return test set if ``test.data`` exists, issues error in case it is not available.
+        fold_id (int, optional): Specifies which fold number to use. Relevant only if train is set to True. The default value of 0 returns the whole data in ``train.data``.
+            Values between 1 and 4 specify, which fold in ``k_fold_cross_val.data`` to use.
+        fold_val (bool, optional): If True, creates dataset using indeces in ``k_fold_cross_val.data`` for for validation part (some odd row) specified by fold_id.
```

Comment:
It looks like the `fold_val` argument kinda takes the role of the `train` argument here? So maybe you could use the `train` argument instead?

Maybe an example usage would be like this:

```python

for fold in range(Abalone.num_folds):
    train_data = Abalone("data", train=True, fold=fold)
    test_data = Abalone("data", train=False, fold=fold)
    ...
```

Reply:
This should work with the revised code.

Reply:
So now fold_val is gone but there is a similar argument that indicates whether we perform hyperparameter search or no